src/Control/Effect/NonDet.hs

```diff
@@ -25,10 +25,10 @@ runNonDet = runAltC . interpret
 newtype AltC f m a = AltC { runAltC :: m (f a) }
 
 instance (Alternative f, Monad f, Traversable f, Carrier sig m, Effect sig, Applicative m) => Carrier (NonDet :+: sig) (AltC f m) where
-  gen a = AltC (gen (pure a))
-  alg = algND \/ (AltC . alg . handle (pure ()) (fmap join . traverse runAltC))
-    where algND Empty      = AltC (gen empty)
-          algND (Choose k) = AltC (liftA2 (<|>) (runAltC (k True)) (runAltC (k False)))
+  ret a = AltC (ret (pure a))
+  eff = AltC . (alg \/ eff . handle (pure ()) (fmap join . traverse runAltC))
```

Comment:
I’ve also been more disciplined about observing the precedence of `\/` relative to `.`, which further reduces the noise.